### PR TITLE
Switch branchprotector and label_sync to run as periodic ProwJobs.

### PR DESCRIPTION
**Pull request by cjwagner**
Hold so that I can delete the existing cronjobs and monitor the new ProwJobs when this merged.
/hold
/cc @stevekuznetsov @fejta @krzyzacy 
/area label_sync
/area prow/branchprotector

### CODE REVIEW

File: config/jobs/kubernetes/test-infra/test-infra-trusted.yaml [hunk 1]
```diff
@@ -252,3 +252,63 @@ periodics:
...
+      - --job-config-path=config
+      - --github-token-path=/etc/github/oauth
+      - --confirm
+      - --github-endpoint=http://ghproxy.default.svc.cluster.local
+      - --github-endpoint=https://api.github.com
+      volumeMounts:
+      - name: oauth
+        mountPath: /etc/github
+        readOnly: true
+    restartPolicy: Never
```
Comment (krzyzacy) [inline config/jobs/kubernetes/test-infra/test-infra-trusted.yaml]: do we need `restartPolicy` anymore? since the lifecycle will be controlled by prow now

Comment (cjwagner) [reply config/jobs/kubernetes/test-infra/test-infra-trusted.yaml]: Nope, we don't need it any more. Looks like Prow forces this value when creating pods for PJs: https://github.com/kubernetes/test-infra/blob/fbc764e23acba94e57ebb15812dbebbbc6a9050b/prow/pod-utils/decorate/podspec.go#L156
I'll remove this.

**Comment by cjwagner** [comment]
Please restamp and I'll give this a go.
Note that `label_sync` won't work until there is a change to `labels.yaml` to trigger the configmap to be uploaded to the `test-pods` namespace:
https://github.com/kubernetes/test-infra/pull/11841/files#diff-ae83e55ccb05896d5229df577d34255dR266

I noticed that we're missing the `area/prow/config-bootstrapper` label though, so I'll just make a PR to add that label to labels.yaml once this merges. https://github.com/kubernetes/test-infra/blob/fbc764e23acba94e57ebb15812dbebbbc6a9050b/prow/cmd/config-bootstrapper/OWNERS#L6

**Comment by cjwagner** [comment]
Cronjobs deleted.

**Comment by cjwagner** [comment]
Both jobs appear to be running properly now:
https://prow.k8s.io/?job=ci-test-infra-branchprotector
https://prow.k8s.io/?job=ci-test-infra-label-sync

**Pull request merged**